vmdb/spec/requests/api/conditions_spec.rb

```diff
@@ -0,0 +1,132 @@
+#
+# REST API Request Tests - Conditions
+#
+# Condition primary collection:
+#   /api/conditions
+#
+# Condition subcollection:
+#   /api/policies/:id/conditions
+#
+require 'spec_helper'
+
+describe ApiController do
+
+  include Rack::Test::Methods
+
+  before(:each) do
+    init_api_spec_env
+  end
+
+  def app
+    Vmdb::Application
+  end
+
+  def create_conditions(count)
+    1.upto(count) do |i|
+      FactoryGirl.create(:condition, :name => "condition_#{i}", :description => "Condition #{i}")
+    end
+  end
+
+  context "Condition collection" do
+    it "query invalid collection" do
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      @success = run_get "#{@cfme[:conditions_url]}/999999"
+
+      expect(@code).to eq(404)
+    end
+
+    it "query conditions with no conditions defined" do
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      @success = run_get @cfme[:conditions_url]
+
+      expect(@code).to eq(200)
+      expect(@result).to have_key("name")
+      expect(@result["name"]).to eq("conditions")
+      expect(@result["resources"].size).to eq(0)
+    end
+
+    it "query conditions" do
+      basic_authorize @cfme[:user], @cfme[:password]
+
+      create_conditions(10)
```

Comment:
Why do you need so many...It's a lot of database interaction...how about just 2 or 3?